codetf.schema.json

```diff
@@ -0,0 +1,228 @@
+        "references": {
+          "type": "array",
+          "items": { "$ref": "#/definitions/reference" }
+        },
+        "properties": {
+          "type": "object",
+          "description": "Vendor-specific properties to help storytelling",
+          "additionalProperties": true
+        },
+        "failedFiles": {
+          "type": "array",
+          "description": "A set of file paths for files that the tool failed to parse or transform",
+          "items": { "type": "string" }
+        },
+        "changeset": {
+          "type": "array",
+          "description": "The changes for a given codemod",
+          "items": { "$ref": "#/definitions/changeset" },
+          "minItems": 0
+        }
+      },
+      "required": ["codemod", "summary", "description", "changeset"]
+    },
+
+    "sarif": {
+      "type": "object",
+      "properties": {
+        "artifact": {
+          "type": "string",
+          "description": "The file containing the SARIF or other input"
+        },
+        "sha1": {
+          "type": "string",
+          "description": "SHA1 hash representing the artifact"
+        }
+      },
+      "required": ["artifact", "sha1"]
+    },
+
+    "reference": {
+      "type": "object",
+      "properties": {
+        "url": {
+          "type": "string",
+          "description": "The URL of the reference"
+        },
+        "description": {
+          "type": "string",
+          "description": "A description of the reference"
+        }
+      },
+      "required": ["url"]
+    },
+
+    "changeset": {
+      "type": "object",
+      "properties": {
+        "path": {
+          "type": "string",
+          "description": "The path of the file (relative to the directory) that was changed"
+        },
+        "diff": {
+          "type": "string",
+          "description": "The unified diff format of the changes made to the file"
+        },
+        "changes": {
+          "type": "array",
+          "items": { "$ref": "#/definitions/change" },
+          "minItems": 1
+        }
+      },
+      "required": ["path", "diff", "changes"]
+    },
+
+    "change": {
+      "type": "object",
+      "properties": {
+        "lineNumber": {
+          "type": "string",
+          "description": "The line number that was changed"
+        },
+        "description": {
+          "type": "string",
+          "description": "A human-readable description of the change"
+        },
+        "diffSide": {
+          "type": "string",
+          "description": "The side of the diff that the change is on. 'right' indicates that the lineNumber and description correspond to a position in the new or modified code. 'left' indicates these fields correspond to a position in the original code. Descriptions of the change itself should generally use 'right', except in cases where the only change is a deletion. In contrast, 'left' can be used to describe the original problem being fixed.",
+          "enum": ["left", "right"]
```

Comment:
Some CodeTF examples I have use UPPERCASE for enums. This appears to be case sensitive.

Reply:
Cool I'll see if I can configure that.